src/Zip Tests/WinZipAesTests.cs

```diff
@@ -611,6 +612,226 @@ public void WZA_CreateZip_NoCompression()
+            TestContext.WriteLine("  Passwords:   {0}", string.Join(", ", passwords));
+            int entries = _rnd.Next(21) + 5;
+
+            string filename = null;
+            var checksums = new Dictionary<string, string>();
+            using (ZipFile zip1 = new ZipFile())
+            {
+                zip1.Encryption = EncryptionAlgorithm.WinZipAes256;
+                zip1.CompressionLevel = Ionic.Zlib.CompressionLevel.None;
+
+                for (int i = 0; i < entries; i++)
+                {
+                    zip1.Password = passwords[i % 3];
+                    if (_rnd.Next(2) == 1)
+                    {
+                        filename = Path.Combine(TopLevelDir, String.Format("Data{0}.bin", i));
+                        int filesize = _rnd.Next(144000) + 5000;
+                        TestUtilities.CreateAndFillFileBinary(filename, filesize);
+                    }
+                    else
+                    {
+                        filename = Path.Combine(TopLevelDir, String.Format("Data{0}.txt", i));
+                        int filesize = _rnd.Next(144000) + 5000;
+                        TestUtilities.CreateAndFillFileText(filename, filesize);
+                    }
+                    zip1.AddFile(filename, "");
+
+                    var chk = TestUtilities.ComputeChecksum(filename);
+                    checksums.Add(Path.GetFileName(filename), TestUtilities.CheckSumToString(chk));
+                }
+
+                zip1.Comment = String.Format("This archive uses Encryption({0}) passwords({1}) no compression.", zip1.Encryption, string.Join(", ", passwords));
+                zip1.Save(zipFileToCreate);
+            }
+
+            // validate all the checksums
+            using (ZipFile zip2 = ZipFile.Read(zipFileToCreate))
+            {
+                int i = 0;
+                foreach (ZipEntry e in zip2)
+                {
+                    if (!e.IsDirectory)
+                    {
+                        Assert.AreEqual<short>(0, (short)e.CompressionMethod);
+                        e.ExtractWithPassword("unpack", passwords[i % 3]);
+                        string PathToExtractedFile = Path.Combine("unpack", e.FileName);
+                        Assert.IsTrue(checksums.ContainsKey(e.FileName));
+
+                        // verify the checksum of the file is correct
+                        string expectedCheckString = checksums[e.FileName];
+                        string actualCheckString = TestUtilities.CheckSumToString(TestUtilities.ComputeChecksum(PathToExtractedFile));
+                        Assert.AreEqual<String>(expectedCheckString, actualCheckString, "Unexpected checksum on extracted filesystem file ({0}).", PathToExtractedFile);
+                        i++;
+                    }
+                }
+            }
+        }
+
+        [TestMethod]
+        public void WZA_CreateZip_MultipleEntriesDifferentEncrytion()

```

Comment:
Typo on the line here - should be WZA_CreateZip_MultipleEntriesDifferentEncryption